lib/puppet/transaction/event_manager.rb

```diff
@@ -147,19 +147,19 @@ def process_callback(resource, callback, events)
     resource.send(callback)
 
     if not resource.is_a?(Puppet::Type.type(:whit))
-      resource.notice "Triggered '#{callback}' from #{events.length} events"
+      resource.notice n_("Triggered '%{callback}' from %{count} event", "Triggered '%{callback}' from %{count} events", events.length) % { :count => events.length, callback: callback }
     end
     return true
   rescue => detail
-    resource.err "Failed to call #{callback}: #{detail}"
+    resource.err _("Failed to call %{callback}: %{detail}") % { callback: callback, detail: detail }
 
     transaction.resource_status(resource).failed_to_restart = true
     resource.log_exception(detail)
     return false
   end
 
   def process_noop_events(resource, callback, events)
-    resource.notice "Would have triggered '#{callback}' from #{events.length} events"
+    resource.notice n_("Would have triggered '%{callback}' from %{count} event", "Would have triggered '%{callback}' from %{count} events", events.length) % { :count => events.length, callback: callback }
```

Comment:
nit: This hash uses two different hash syntaxes for different keys - should probably just use one (seems like the newer form is dominant here)